lib/ApiRoute.js

```diff
@@ -37,14 +53,32 @@ class ApiRoute {
   }
 
   /**
-   * converts class-based route handlers to Next API handler
-   * @returns {ApiHandler}
+   * @override
    */
   as_handler() {
-    return (req, res) => {
+    const allowedMethods = ['post', 'get', 'delete', 'patch'];
+
+    return async (req, res) => {
       const defaultHandler = (_, res, __) => res.status(404).end();
-      if (req.method?.toLowerCase() === 'as_handler') return defaultHandler;
-      return (this[req.method?.toLowerCase()] ?? defaultHandler)(req, res, prisma);
+      const method = req.method?.toLowerCase();
+
+      if (!allowedMethods.includes(method)) {
+        return defaultHandler;
+      } else if (!this.options?.protect?.[method]) {
```

Comment:
Revert these changes except the allowedMethods check and setting req.session. We don't need protected routes because everything is locked down by the middleware. Later on, if we want to have pages/routes be public, we can change the middleware config: https://next-auth.js.org/tutorials/securing-pages-and-api-routes#nextjs-middleware.